databricks/koalas/namespace.py

```diff
@@ -538,6 +540,216 @@ def column_name(value):
+    0    a
+    1    b
+    2    c
+    3    d
+    Name: 0, dtype: object
+
+    Combine two ``DataFrame`` objects with identical columns.
+
+    >>> df1 = ks.DataFrame([['a', 1], ['b', 2]],
+    ...                    columns=['letter', 'number'])
+    >>> df1
+      letter  number
+    0      a       1
+    1      b       2
+    >>> df2 = ks.DataFrame([['c', 3], ['d', 4]],
+    ...                    columns=['letter', 'number'])
+    >>> df2
+      letter  number
+    0      c       3
+    1      d       4
+    >>> ks.concat([df1, df2])
+      letter  number
+    0      a       1
+    1      b       2
+    0      c       3
+    1      d       4
+
+    Combine ``DataFrame`` and ``Series`` objects with different columns.
+
+    >>> ks.concat([df2, s1, s2])
+          0 letter  number
+    0  None      c     3.0
+    1  None      d     4.0
+    0     a   None     NaN
+    1     b   None     NaN
+    0     c   None     NaN
+    1     d   None     NaN
+
+    Combine ``DataFrame`` objects with overlapping columns
+    and return everything. Columns outside the intersection will
+    be filled with ``None`` values.
+
+    >>> df3 = ks.DataFrame([['c', 3, 'cat'], ['d', 4, 'dog']],
+    ...                    columns=['letter', 'number', 'animal'])
+    >>> df3
+      letter  number animal
+    0      c       3    cat
+    1      d       4    dog
+    >>> ks.concat([df1, df3])
+      animal letter  number
+    0   None      a       1
+    1   None      b       2
+    0    cat      c       3
+    1    dog      d       4
+
+    Combine ``DataFrame`` objects with overlapping columns
+    and return only those that are shared by passing ``inner`` to
+    the ``join`` keyword argument.
+
+    >>> ks.concat([df1, df3], join="inner")
+      letter  number
+    0      a       1
+    1      b       2
+    0      c       3
+    1      d       4
+    """
+    if not isinstance(objs, (dict, Iterable)):
+        raise TypeError('first argument must be an iterable of koalas '
+                        'objects, you passed an object of type '
+                        '"{name}"'.format(name=type(objs).__name__))
+
+    if all(map(lambda obj: obj is None, objs)):
+        raise ValueError("All objects passed were None")
+    objs = list(filter(lambda obj: obj is not None, objs))
+
+    for obj in objs:
+        if not isinstance(obj, (Series, DataFrame)):
+            raise TypeError('cannot concatenate object of type '"'{name}"'; only ks.Series '
+                            'and ks.DataFrame are valid'.format(name=type(objs).__name__))
+
+    # Series, Series ...
+    should_return_series = all(map(lambda obj: isinstance(obj, Series), objs))
+
+    # DataFrame, Series ... & Series, Series ...
+    new_objs = []
+    for obj in objs:
+        if isinstance(obj, Series):
+            obj = obj.to_dataframe()
+        new_objs.append(obj)
+    objs = new_objs
+
+    # DataFrame, DataFrame, ...
+    indices_of_kdfs = list(map(lambda kdf: kdf._metadata.index_map, objs))
```

Comment:
just out of curiosity, is this the same as `[kdf._metadata.index_map for kdf in objs]`?

Reply:
oh, yes. that looks simpler.